src/Altinn.App.Core/Features/Payment/Processors/Nets/NetsPaymentProcessor.cs

```diff
@@ -0,0 +1,195 @@
+        };
+
+        HttpApiResult<NetsCreatePaymentSuccess> httpApiResult = await _netsClient.CreatePayment(payment);
+        if (!httpApiResult.IsSuccess || httpApiResult.Result?.HostedPaymentPageUrl is null)
+        {
+            throw new PaymentException(
+                "Failed to create payment\n" + httpApiResult.Status + " - " + httpApiResult.RawError
+            );
+        }
+
+        string hostedPaymentPageUrl = httpApiResult.Result.HostedPaymentPageUrl;
+        string paymentId = httpApiResult.Result.PaymentId;
+
+        return new PaymentDetails
+        {
+            PaymentId = paymentId,
+            RedirectUrl = AddLanguageQueryParam(hostedPaymentPageUrl, language)
+        };
+    }
+
+    /// <inheritdoc />
+    public async Task<bool> TerminatePayment(Instance instance, PaymentInformation paymentInformation)
+    {
+        if (paymentInformation.PaymentDetails?.PaymentId is null)
+        {
+            throw new PaymentException("PaymentId is missing in paymentInformation. Can't terminate.");
+        }
+
+        bool result = await _netsClient.TerminatePayment(paymentInformation.PaymentDetails.PaymentId);
+        return result;
+    }
+
+    /// <inheritdoc />
+    public async Task<(PaymentStatus status, PaymentDetails paymentDetails)> GetPaymentStatus(
+        Instance instance,
+        string paymentId,
+        decimal expectedTotalIncVat,
+        string? language
+    )
+    {
+        HttpApiResult<NetsPaymentFull> httpApiResult = await _netsClient.RetrievePayment(paymentId);
+
+        if (!httpApiResult.IsSuccess || httpApiResult.Result is null)
+        {
+            throw new PaymentException(
+                "Failed to retrieve payment\n" + httpApiResult.Status + " - " + httpApiResult.RawError
+            );
+        }
+
+        NetsPayment payment = httpApiResult.Result.Payment!;
+        decimal? chargedAmount = payment.Summary?.ChargedAmount;
+
+        PaymentStatus status = chargedAmount > 0 ? PaymentStatus.Paid : PaymentStatus.Created;
+        NetsPaymentDetails? paymentPaymentDetails = payment.PaymentDetails;
+
+        PaymentDetails paymentDetails =
+            new()
+            {
+                PaymentId = paymentId,
+                RedirectUrl = AddLanguageQueryParam(payment.Checkout!.Url!, language),
+                Payer = NetsMapper.MapPayerDetails(payment.Consumer),
+                PaymentType = paymentPaymentDetails?.PaymentType,
+                PaymentMethod = paymentPaymentDetails?.PaymentMethod,
+                CreatedDate = payment.Created,
+                ChargedDate = payment.Charges?.FirstOrDefault()?.Created,
+                InvoiceDetails = NetsMapper.MapInvoiceDetails(paymentPaymentDetails?.InvoiceDetails),
+                CardDetails = NetsMapper.MapCardDetails(paymentPaymentDetails?.CardDetails),
+            };
+
+        return (status, paymentDetails);
+    }
+
+    private static string AddLanguageQueryParam(string url, string? language)
```

Comment:
In https://github.com/Altinn/app-lib-dotnet/pull/635 we use a standard library function to add to query string.